src/Uno.UI/UI/Xaml/Controls/MediaPlayer/HtmlMediaPlayer.wasm.cs

```diff
@@ -0,0 +1,519 @@
+using System;
+using System.IO;
+using Windows.Foundation;
+using Windows.UI.Xaml.Media;
+using Uno.Extensions;
+using Uno.Foundation;
+using Uno.Foundation.Logging;
+using Windows.UI.Xaml.Media.Imaging;
+using Uno.Disposables;
+using Windows.Storage.Streams;
+using System.Runtime.InteropServices;
+using Uno.UI.Xaml.Media;
+using Windows.UI;
+using System.Linq;
+using Windows.Media.Playback;
+using Windows.UI.Xaml;
+
+namespace Windows.UI.Xaml.Controls;
+
+internal class HtmlMediaPlayer : Border, IHtmlMediaPlayer, IDisposable
+{
+	private static readonly HtmlVideo _htmlVideo = new HtmlVideo();
+	private static readonly HtmlAudio _htmlAudio = new HtmlAudio();
+	private string[] audioTagAllowedFormats = { ".MP3", ".WAV" };
+	private string[] videoTagAllowedFormats = { ".MP4", ".WEBM", ".OGG" };
+
+	public event EventHandler<object> OnSourceLoaded;
+	public event EventHandler<object> OnSourceFailed;
+	public event EventHandler<object> OnSourceEnded;
+	public event EventHandler<object> OnMetadataLoaded;
+	public event EventHandler<object> OnTimeUpdate;
+
+	public HtmlMediaPlayer()
+	{
+		_htmlVideo.SetStyle("visibility", "hidden");
+		_htmlAudio.SetStyle("visibility", "hidden");
+		ConfigureMetadata(true);
+		ConfigureMetadata(false);
+
+		SourceLoaded += OnHtmlSourceLoaded;
+		SourceFailed += OnHtmlSourceFailed;
+		SourceEnded += OnHtmlSourceEnded;
+		MetadataLoaded += OnHtmlMetadataLoaded;
+		TimeUpdated += OnHtmlTimeUpdated;
+
+		WebAssemblyRuntime.InvokeJS($"console.log('Adding Media elements');");
```

Comment:
We must not use InvokeJS anymore, but rather JSImport/JSExport (see existing uses in Uno as examples). We'll be making this API forbidden very soon to be CSP compliant.